packages/cli/package.json

```diff
@@ -1,103 +1,106 @@
+    "files": [
+        "lib",
+        "scripts"
     ],
-    "collectCoverage": false,
-    "reporters": [
-      "default",
-      "jest-junit",
-      "jest-stare",
-      "jest-html-reporter"
-    ],
-    "coverageReporters": [
-      "json",
-      "lcov",
-      "text",
-      "cobertura"
-    ],
-    "coverageDirectory": "<rootDir>/__tests__/__results__/unit/coverage"
-  },
-  "jest-stare": {
-    "resultDir": "<rootDir>/__tests__/__results__/jest-stare"
-  },
-  "jest-junit": {
-    "outputDirectory": "<rootDir>/__tests__/__results__/unit/junit"
-  },
-  "jest-html-reporter": {
-    "pageTitle": "Kubernetes secrets Plugin for Zowe CLI Test Results",
-    "outputPath": "<rootDir>/__tests__/__results__/results.html",
-    "includeFailureMsg": true
-  }
+    "main": "lib/index.js",
+    "typings": "lib/index.d.ts",
+    "imperative": {
+        "configurationModule": "lib/imperative/configuration.js"
+    },
+    "scripts": {
+        "clean": "rimraf lib",
+        "checkTestsCompile": "echo \"Checking that test source compiles...\" && tsc --project __tests__/test-tsconfig.json --noEmit ",
+        "circularDependencyCheck": "madge -c lib",
+        "lint": "eslint \"src/**/*.ts\" \"**/__tests__/**/*.ts\"",
+        "lint:src": "eslint \"src/**/*.ts\" --ignore-pattern \"**/__tests__/**/*.ts\"",
+        "lint:tests": "eslint \"**/__tests__/**/*.ts\"",
+        "prebuild": "npm run clean && npm run lint && echo Using TypeScript && tsc --version",
+        "build": "tsc --pretty && npm run checkTestsCompile && npm run circularDependencyCheck",
+        "prepublishOnly": "npm run build && bash ./scripts/bundleKeytar.sh",
+        "watch": "tsc --pretty --watch",
+        "watch:test": "jest --watch",
+        "installPlugin": "npm install && npm run clean && npm run build && zowe plugins install .",
+        "postuninstall": "echo \"CLI Profiles that you have created will need to be recreated because their credentials are stored in the credential manager that has been uninstalled. Refer to the available options in the help text to create a new profile 'zowe profiles create zosmf-profile -h'.\" 1>&2",
+        "typedoc": "typedoc --out ./docs/typedoc/ ./src/ --disableOutputCheck",
+        "test": "npm run test:unit && npm run test:system",
+        "test:cleanResults": "rimraf __tests__/__results__",
+        "test:unit": "env-cmd -f __tests__/__resources__/env/unit.env jest --coverage --testPathIgnorePatterns \".*/__system__/.*\" ",
+        "test:system": "env-cmd -f __tests__/__resources__/env/system.env jest .*/__system__/.* --coverage false --runInBand",
+        "coverage": "jest --coverage",
+        "package": "npm run clean && node ../../scripts/updateLicenses.js && npm run build && npm pack --pack-destination ../../dist"
+    },
+    "peerDependencies": {
+        "@zowe/cli": "^7.13.0",
+        "@zowe/imperative": "*",
+        "@zowe/cli-test-utils": "^7.13.0"
+    },
```

Comment:
Per the Zowe CLI conformance criteria, only Imperative should be listed as a peer dep:
> Plug-ins must not have an @zowe/cli peer dependency for improved npm@7 compatibility. The only peer dependencies that should be included are
packages which are imported in the plug-in's source code (e.g., @zowe/imperative).

The `@zowe/cli` and `@zowe/cli-test-utils` could be moved to "devDependencies".

For the `@zowe/imperative` peer dep, I think we'd want to require version 5.10.0 (which introduced support for plugin lifecycles) or newer.

Reply:
That's sounds good to me.

For `@zowe/imperative` it is defined as `*` so it pulls the version from it's parent `package.json` this being the one in the root of the monorepo which defines `"@zowe/imperative": "^5.12.0"`

Reply:
> For @zowe/imperative it is defined as * so it pulls the version from it's parent package.json this being the one in the root of the monorepo which defines "@zowe/imperative": "^5.12.0"

The `*` will resolve correctly in the monorepo, but the CLI plugin may be bundled into a TGZ and installed independently where `*` won't be resolved. I think it would be good to have the version number specified in both places, even though I know it seems kind of redundant, because then if someone tries to install the plugin into a version of the CLI that is too old, they will see a warning about the Imperative version being potentially incompatible.